doc/source/conf.py

```diff
@@ -122,67 +177,75 @@
 htmlhelp_basename = project + 'doc'
 
 
-# -- Options for LaTeX output -------------------------------------------------
+# -- Options for LaTeX output --------------------------------------------------
 
 # Grouping the document tree into LaTeX files. List of tuples
 # (source start file, target name, title, author, documentclass [howto/manual]).
 latex_documents = [('index', project + '.tex', project + u' Documentation',
                     author, 'manual')]
 
 
-# -- Options for manual page output -------------------------------------------
+# -- Options for manual page output --------------------------------------------
 
 # One entry per manual page. List of tuples
 # (source start file, name, description, authors, manual section).
 man_pages = [('index', project.lower(), project + u' Documentation',
               [author], 1)]
 
 # -- Swap to Napoleon ---------------------------------------------------------
-
-extensions.remove('astropy_helpers.sphinx.ext.numpydoc')
 extensions.append('sphinx.ext.napoleon')
-
 # Disable having a separate return type row
 napoleon_use_rtype = False
 # Disable google style docstrings
 napoleon_google_docstring = False
 
+# -- Options for the edit_on_github extension ----------------------------------------
+extensions.remove('astropy_helpers.extern.numpydoc')
+extensions.append('sphinx.ext.napoleon')
 
-# -- Options for the edit_on_github extension ---------------------------------
-extensions += ['astropy_helpers.sphinx.ext.edit_on_github',
-               'sphinx.ext.doctest']
-
+# Disable having a separate return type row
+napoleon_use_rtype = False
+# Disable google style docstrings
+napoleon_google_docstring = False
+extensions += ['astropy_helpers.sphinx.ext.edit_on_github', 'sphinx.ext.doctest']
 
 # Don't import the module as "version" or it will override the
 # "version" configuration parameter
-# TODO: make this smart like astropy
+from sunpy import version as versionmod
 edit_on_github_project = "sunpy/sunpy"
-edit_on_github_branch = "master"
-
+if versionmod.release:
+    edit_on_github_branch = "v{0}.{1}.x".format(
+        versionmod.major, versionmod.minor)
+else:
+    edit_on_github_branch = "master"
 edit_on_github_source_root = ""
 edit_on_github_doc_root = "docs"
 
+edit_on_github_skip_regex = '_.*|api/.*'
```

Comment:
Ah ha! This looks like it might be able to fox #2169 maybe if this was changed to 'generated' over 'api'?